File: AnkiDroid/src/main/java/com/ichi2/upgrade/Upgrade.java

package com.ichi2.upgrade;

import com.ichi2.libanki.Collection;

import com.ichi2.utils.JSONException;
import com.ichi2.utils.JSONObject;

import timber.log.Timber;

public class Upgrade {

    public static boolean upgradeJSONIfNecessary(Collection col, JSONObject conf, String name, boolean defaultValue) {
        boolean val = defaultValue;
        try {
            val = conf.getBoolean(name);
        } catch (JSONException e) {
            Timber.w(e);
            // workaround to repair wrong values from older libanki versions
            try {
                conf.put(name, val);
            } catch (JSONException e1) {
                Timber.w(e1);
                // do nothing
            }
            col.save();
        }
        return val;
    }
}

File: AnkiDroid/src/main/java/com/ichi2/upgrade/Upgrade.kt
//noinspection MissingCopyrightHeader #8659
package com.ichi2.upgrade

import com.ichi2.libanki.Collection
import org.json.JSONException
import timber.log.Timber

/**
 * Ensure the configuration [name] is set to a value that can be cast to a Boolean.
 * @param name the configuration key's name
 * @return current value if it exists and can be cast to Boolean, or default value
 */
fun Collection.upgradeJSONIfNecessary(name: String, defaultValue: Boolean) =
    try {
        get_config_boolean(name)
    } catch (e: JSONException) {
        Timber.w(e)
        // workaround to repair wrong values from older libanki versions
        try {
            set_config(name, defaultValue)
        } catch (e1: JSONException) {
            Timber.w(e1)
            // do nothing
        }
        defaultValue
    }
